userspace/libsinsp/threadinfo.h

```diff
@@ -57,6 +57,25 @@ struct erase_fd_params {
  *  @{
  */
 
+/*!
+  \brief Container holding parameters to be provided to sinsp_threadinfo constructor.
+  An instance of this struct is meant to be shared among all sinsp_threadinfo instances.
+*/
+struct sinsp_threadinfo_ctor_params {
+	// The following fields are externally provided and access to them is expected to be
+	// read-only.
+	const sinsp_network_interfaces& network_interfaces;
+	const sinsp_fdinfo_factory& fdinfo_factory;
+	const sinsp_fdtable_factory& fdtable_factory;
+	const std::shared_ptr<libsinsp::state::table_entry::dynamic_struct::field_infos>&
+	        thread_manager_dyn_fields;
+
+	// The following fields are externally provided and expected to be populated/updated by the
+	// thread info.
+	std::shared_ptr<sinsp_thread_manager>& thread_manager;
+	std::shared_ptr<sinsp_usergroup_manager>& usergroup_manager;
+};
+
```

Comment:
I know we already discussed the reason why this has been moved out of `sinsp_threadinfo`, but could you add a comment here specifying why for future reference? :smile:

Reply:
Oh yes, I'm sorry, I didn't notice it. Fine by me then :smile: